tools/tck-api/src/main/scala/org/opencypher/tools/tck/api/Scenario.scala

```diff
@@ -40,11 +40,11 @@ import scala.compat.Platform.EOL
 import scala.language.implicitConversions
 import scala.util.{Failure, Success, Try}
 
-case class Scenario(featureName: String, name: String, tags: Set[String], steps: List[Step], source: Pickle) {
+case class Scenario(featureName: String, name: String, category: List[String], tags: Set[String], steps: List[Step], source: Pickle) {
```

Comment:
The list encodes the hierarchy. Any scenario from `clauses/match/Match1.feature` will have a list `List("clauses", "match")`. The directory name along the directory path. That is why 
(a) it is inherently ordered (and I do not see why this information should be thrown away), 
(b) it could potentially contain duplicates (like a directory can contain a directory of the same name), 
(c) it is named singular, since you can think of the directory path/the list as the full qualified name of the single category in which the scenario lives (although, I can also easily live with `categories`, if your opinion still holds).

As more background, my thinking is that the feature files are the truth. The API simple reflects that truth into Scala object of class `Scenario` for programatic consumption. With categorizations, we add meaningful directories to the truth of the feature files. The changes of this PR shall reflect that without restricting what can be done in the truth. 

A scenario cannot be in two categories. For that to be the case, it would have to be in two separate feature files located in two separate directories. Assuming that every feature file contains a uniquely named feature, the API will not create twice the same `Scenario` object even without my change.

Finally, the intention is to use tags for cross-cutting concerns (e.g. which parts of the language are used by a scenario) instead of putting scenarios into multiple categories. The tooling for tags already exists. We just have to start to using them in the feature files. Still the categories are useful. They help to focus thinking about what actually a scenario is suppose to test, even though it may use a number of different parts of the language.

For all this thinking, I am reluctant adopt your suggestion here; apart from the plural for the parameter/fields. Please indicate how strongly you feel about the plural.

Reply:
cf. Scenario.scala:43

Reply:
OK, that makes sense. I have two thoughts then: I think plural is correct because of the type of the field (a collection). If you feel like this is awkward, then I suggest modelling a new class `ScenarioCategory` or similar which can have a `def categories: List[String]` and then the variable for the `ScenarioCategory` could be singular. This may be a bit overkill, however.

Reply:
Okay. I have changed the variable and parameter name to the plural.